chevah/compat/testing/testcase.py

```diff
@@ -1006,9 +1009,16 @@ def addCleanup(self, function, *args, **kwargs):
     def callCleanup(self):
         """
         Call all cleanup methods.
+
+        If a cleanup fails, the next cleanups will continue to be called and
+        the first failure is raised.
         """
         for function, args, kwargs in reversed(self.__cleanup__):
-            function(*args, **kwargs)
+            try:
+                function(*args, **kwargs)
+            except Exception as error:  # noqa:cover
+                self._teardown_errors.append(error)
```

Comment:
How about appending a `traceback.format_exc()` instead of the error itself?

Reply:
True. Thanks.